src/app/pages/questions/components/question/audio-input/audio-input.component.ts

```diff
@@ -3,112 +3,151 @@ import {
   EventEmitter,
   Input,
   OnChanges,
+  OnDestroy,
   OnInit,
   Output
 } from '@angular/core'
-import { Device } from '@ionic-native/device/ngx'
-import { AlertController, NavController } from 'ionic-angular'
+import { AlertController, NavController, Platform } from 'ionic-angular'
 
-// NOTE: File path to opensmile.js; Adding opensmile plugin
-import * as opensmile from '../../../../../../../plugins/cordova-plugin-opensmile/www/opensmile'
-import { Answer } from '../../../../../shared/models/answer'
+import {
+  MIN_SEC,
+  SEC_MILLISEC
+} from '../../../../../../assets/data/defaultConfig'
+import { LocKeys } from '../../../../../shared/enums/localisations'
 import { Section } from '../../../../../shared/models/question'
+import { TranslatePipe } from '../../../../../shared/pipes/translate/translate'
 import { AndroidPermissionUtility } from '../../../../../shared/utilities/android-permission'
-import { QuestionsPageComponent } from '../../../containers/questions-page.component'
+import { HomePageComponent } from '../../../../home/containers/home-page.component'
 import { AudioRecordService } from '../../../services/audio-record.service'
 
-declare var cordova: any
-declare var window: any
-
 @Component({
   selector: 'audio-input',
   templateUrl: 'audio-input.component.html'
 })
-export class AudioInputComponent implements OnInit, OnChanges {
+export class AudioInputComponent implements OnChanges, OnDestroy, OnInit {
   @Output()
   valueChange: EventEmitter<any> = new EventEmitter<any>()
   @Input()
   sections: Section[]
   @Input()
   currentlyShown: boolean
 
-  filename: string
-  name: string
-  filepath: string
-  recording: boolean
-  value: string = null
-  configFile = 'liveinput_android.conf'
-  compression = 1
-  platform = false
-  answer_b64: string = null
-  permission: any
-
-  answer: Answer = {
-    id: null,
-    value: null,
-    type: 'audio'
-  }
-
-  ngOnInit() {}
-
-  ngOnChanges() {
-    if (this.currentlyShown) {
-      this.startRecording()
-    }
-  }
+  avgWordsPerMinute = 100
+  alertShown = false
+  resumeListener
+  pauseListener
+  recordingTimeout
 
   constructor(
-    public questions: QuestionsPageComponent,
     private audioRecordService: AudioRecordService,
     private permissionUtil: AndroidPermissionUtility,
     public navCtrl: NavController,
     public alertCtrl: AlertController,
-    private device: Device
+    private platform: Platform,
+    private translate: TranslatePipe
   ) {
-    // NOTE: Stop audio recording when application is on pause / backbutton is pressed
-    document.addEventListener('pause', () => {
-      console.log('on pause')
-      // if (this.navCtrl.isActive(this.navCtrl.getActive()) == false) {
-      this.audioRecordService.stopAudioRecording()
-      // }
-    })
+    this.permissionUtil.checkPermissions()
```

Comment:
If this is only for android, it works fine. But I think we should handle the exceptions here. If the permissions for audio is not available, the task should exit with a message or request for permission.